core/arch/arm/mm/mobj.c

```diff
@@ -155,12 +157,6 @@ struct mobj *mobj_phys_alloc(paddr_t pa, size_t size, uint32_t cattr,
 	enum teecore_memtypes area_type;
 	void *va;
 
-	if ((pa & CORE_MMU_USER_PARAM_MASK) ||
```

Comment:
I think I'd rather have another `mobj` type. With that we could use the same physical `mobj` for a references into the shared memory pool.

Another option is to return an offset that should be used with the returned `mobj` from `thread_rpc_alloc()` also, but I think it's clearer with another `mobj` type that hides this.